kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -444,17 +448,30 @@ private static void validateSchemaEvolution(
    * Validates a given schema evolution by using field ID as the source of truth for identifying
    * fields
    */
-  private static void validateSchemaEvolutionById(StructType currentSchema, StructType newSchema) {
+  private static void validateSchemaEvolutionById(
+      StructType currentSchema, StructType newSchema, Set<String> clusteringColumnPhysicalNames) {
     Map<Integer, StructField> currentFieldsById = fieldsById(currentSchema);
     Map<Integer, StructField> updatedFieldsById = fieldsById(newSchema);
     SchemaChanges schemaChanges = computeSchemaChangesById(currentFieldsById, updatedFieldsById);
     validatePhysicalNameConsistency(schemaChanges.updatedFields());
     // Validates that the updated schema does not contain breaking changes in terms of types and
     // nullability
     validateUpdatedSchemaCompatibility(schemaChanges);
+    validateClusteringColumnsNotDropped(
```

Comment:
qn: should we check if the columns are relocated or moved?

Reply:
I think whether a clustering column is reordered is inconsequential insofar as it is not an invalid reordering (like moving it out of its containing struct). That invalid reordering will happen in a separate PR, but clustering columns are just columns for that validation's purpose.